seqr/models.py

```diff
@@ -397,115 +397,68 @@ class Sample(ModelWithGUID):
+    DATASET_TYPE_SPLICE_JUNCTIONS = 'SPLICE'
+    DATASET_TYPE_ASE = 'ASE'
+    DATASET_TYPE_CHOICES = (
+        (DATASET_TYPE_READ_ALIGNMENTS, 'Alignment'),
+        (DATASET_TYPE_VARIANT_CALLS, 'Variant Calls'),
+        (DATASET_TYPE_SV_CALLS, 'SV Calls'),
+        (DATASET_TYPE_BREAKPOINTS, 'Breakpoints'),
+        (DATASET_TYPE_SPLICE_JUNCTIONS, 'Splice Junction Calls'),
+        (DATASET_TYPE_ASE, 'Allele Specific Expression'),
     )
 
-    DATASET_STATUS_QUEUED = 'Q'
-    DATASET_STATUS_LOADING = 'G'
-    DATASET_STATUS_LOADED = 'L'
-    DATASET_STATUS_FAILED = 'F'
-    DATASET_STATUS_CHOICES = (
-        (DATASET_STATUS_QUEUED, 'Queued'),
-        (DATASET_STATUS_LOADING, 'Loading'),
-        (DATASET_STATUS_LOADED, 'Loaded'),
-        (DATASET_STATUS_FAILED, 'Failed'),
-    )
+    SAMPLE_STATUS_IN_SEQUENCING = 'seq'
+    SAMPLE_STATUS_COMPLETED_SEQUENCING = 'seq_done'
+    SAMPLE_STATUS_SEQUENCING_FAILED = 'seq_fail_1'
+    SAMPLE_STATUS_LOADING = 'loading'
+    SAMPLE_STATUS_LOADED = 'loaded'
 
-    name = models.TextField(null=True, blank=True)
-    description = models.TextField(null=True, blank=True)
-    status = models.CharField(null=True, blank=True, db_index=True, max_length=1, choices=DATASET_STATUS_CHOICES)
+    SAMPLE_STATUS_CHOICES = (
+        (SAMPLE_STATUS_IN_SEQUENCING, 'In Sequencing'),
+        (SAMPLE_STATUS_COMPLETED_SEQUENCING, 'Completed Sequencing'),
+        (SAMPLE_STATUS_SEQUENCING_FAILED, 'Failed Sequencing - Abandoned'),
+        (SAMPLE_STATUS_LOADING, 'Loading'),
+        (SAMPLE_STATUS_LOADED, 'Loaded'),
+    )
 
-    genome_version = models.CharField(max_length=5, choices=GENOME_VERSION_CHOICES, default=GENOME_VERSION_GRCh37)
+    individual = models.ForeignKey('Individual', on_delete=models.PROTECT, null=True)
 
-    # When a dataset is copied from source_file_path to an internal seqr database or directory,
-    # the dataset_id should be the pointer used to query this data. Although global uniqueness
-    # is not enforced, the dataset_id value should avoid collisions, and should be derived only from
-    # properties of the dataset itself (eg. creation date, size, or md5) so that if a dataset
-    # is added a second time, it would be assigned the same dataset id as before.
-    # This will allow datasets to be processed and loaded only once, but shared between projects if
-    # needed by using the same dataset_id in the Dataset records of different projects.
-    dataset_id = models.TextField(null=True, blank=True, db_index=True)   # elasticsearch index
+    sample_type = models.CharField(max_length=20, choices=SAMPLE_TYPE_CHOICES, null=True, blank=True)
+    dataset_type = models.CharField(max_length=20, choices=DATASET_TYPE_CHOICES, null=True, blank=True)
 
-    dataset_location = models.TextField(null=True, blank=True, db_index=True)
+    # The sample's id in the underlying dataset (eg. the VCF Id for variant callsets).
+    sample_id = models.TextField(db_index=True)
```

Comment:
should these be unique?

Reply:
it's unique relative to all sample ids in the VCF or elasticsearch index, but not necessarily globally unique